src/JSInterop/Microsoft.JSInterop/src/IJSRuntime.cs

```diff
@@ -24,6 +25,21 @@ public interface IJSRuntime
     /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
     ValueTask<TValue> InvokeAsync<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string identifier, object?[]? args);
 
+    /// <summary>
+    /// Invokes the specified JavaScript function asynchronously.
+    /// <para>
+    /// <see cref="JSRuntime"/> will apply timeouts to this operation based on the value configured in <see cref="JSRuntime.DefaultAsyncTimeout"/>. To dispatch a call with a different timeout, or no timeout,
+    /// consider using <see cref="InvokeAsync{TValue}(string, CancellationToken, object[])" />.
+    /// </para>
+    /// </summary>
+    /// <typeparam name="TValue">The JSON-serializable return type.</typeparam>
+    /// <param name="identifier">An identifier for the function to invoke. For example, the value <c>"someScope.someFunction"</c> will invoke the function <c>window.someScope.someFunction</c>.</param>
+    /// <param name="resolver">The <see cref="IJsonTypeInfoResolver"/> to use for JSON serialization and deserialization.</param>
+    /// <param name="args">JSON-serializable arguments.</param>
+    /// <returns>An instance of <typeparamref name="TValue"/> obtained by JSON-deserializing the return value.</returns>
+    ValueTask<TValue> InvokeAsync<[DynamicallyAccessedMembers(JsonSerialized)] TValue>(string identifier, IJsonTypeInfoResolver resolver, object?[]? args)
+        => throw new InvalidOperationException($"Supplying a custom {nameof(IJsonTypeInfoResolver)} is not supported by the current JS runtime");
+
```

Comment:
It might be the case that we can further optimize WebAssembly startup to eliminate marshalled JS interop altogether, and thus make these APIs less necessary. It would just require that we flatten Blazor's existing internal JSON-serializable data types into argument lists that are compatible with [`[JSImport]`/`[JSExport]`](https://learn.microsoft.com/en-us/aspnet/core/blazor/javascript-interoperability/import-export-interop?view=aspnetcore-8.0#call-javascript-from-net). For Blazor Server, we could either continue to use the existing types or use the shared flattened arguments with the remote JS runtime.

However, it still may be valuable to make these APIs available to customers even if they don't get used internally within the framework.

Either way, I plan on continuing to experiment and see whether it's feasible to only use unmarshalled interop during WebAssembly startup.